c/src/core/codec.c

```diff
@@ -364,37 +364,72 @@ static int pn_data_inspect(void *obj, pn_string_t *dst)
 #define pn_data_hashcode NULL
 #define pn_data_compare NULL
 
-pn_data_t *pn_data(size_t capacity)
+pn_data_t *pni_data(size_t capacity, size_t initial_buffer)
```

Comment:
That is probably just an artefact of the evolution of the patch. Initialy I began by allocating an empty buffer rather than 64 bytes, but then went further. I'll remove the extra function.

Reply:
Thinking further I'm reasonably convinced that always making it lazy shouldn't be an issue as we now make the buff only when we need it - so if we always make at least the previous 64 byte size then we do strictly no more work and if we delay the creation we can be avoid making a 64 byte buffer only to immediately expand it due to the size of the first interned data